Source/PINMemoryCache.m

```diff
@@ -196,6 +200,27 @@ - (void)trimMemoryToDate:(NSDate *)trimDate
     }
 }
 
+- (void)removeExpiredObjects
```

Comment:
Sorry this comment is in the wrong place (you're using dates[key] below which gets updated on access in objectForKey), but, should objectForKey: not set the access date if an ageLimit is set?

Reply:
Hmm, good point. The disk cache checks if object && !self->_ttlCache before setting the access date. This logic currently exists; I wonder if that's a bug. Shall I update the memory cache to do the same check?

Reply:
I'm wondering which is actually the bug…

`trimDiskToSizeByDate` and `trimToCostLimitByDate` both use last access to trim to order which objects they trim first. If ttlCache is enabled or there are age limits, objects which exceed the age limit should be trimmed first and then objects should be trimmed by last access. Does that seem right? In which case the two methods will need to be updated and the diskCache will need to remove its check of `object && !self->_ttlCache`…

Reply:
Made the updates you suggested. PTAL.